src/firecracker/src/api_server_adapter.rs

```diff
@@ -245,7 +249,8 @@ pub(crate) fn run_with_api(
     api_thread.join().expect("Api thread should join");
 
     match result {
-        Ok(exit_code) => Err(ApiServerError::MicroVMStoppedWithoutError(exit_code)),
+        Ok(Ok(())) => Ok(()),
+        Ok(Err(exit_code)) => Err(ApiServerError::MicroVMStoppedWithoutError(exit_code)),
```

Comment:
Can we avoid the `Result<Result<...>,...>` by instead keeping `run_microvm` as is today (e.g returning `FcExitCode` instead of `Result<(), FcExitCode>`), and updating this match to be 
```rs
Ok(FcExitCode::Ok) => Ok(()),
Ok(exit_code) => Err(ApiServerError::MicroVMStoppedWithoutError(exit_code))
Err(exit_error) => Err(exit_error)
```

Reply:
If we followed this pattern for all functions, functions would only return `ExitCode` and we would see these matches throughout the code base. I think this is the wrong direction. Functions should align with Rust practices for error handling by returning `Result`.

Reply:
That's fair